frontend/src/utils/PublicRoute.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import jwt_decode from 'jwt-decode';
+
+const PublicRoute = ({ component: Component, ...rest }) => (
+	<Route
+		{...rest}
+		render={props => {
+			if (localStorage.token) {
+				if (
+					jwt_decode(localStorage.token) &&
+					jwt_decode(localStorage.token).exp > Math.floor(Date.now() / 1000) // checks if token is a JWT and if it is expired
+				) {
+					return <Redirect to="/dashboard" />;
```

Comment:
Why does an expired token cause a public route to redirect to the dashboard which is a private route? Especially since our only public route is the landing page?

Reply:
Ah I see. With the comment I read it as “if there’s a token and it’s expired redirect to dashboard”.